### PR TITLE
Upgrade dotnet versions

### PR DESCRIPTION
**Pull request by Mandur**
* Upgrade .net 2.2 to 3.1
* Upgrade function runtime to ~3
* Use standard Arduino library for the CI

### CODE REVIEW

**Comment by TechPreacher** [comment]
Should we also update the .NET and dependencies versions in the \Samples and \Tools project folder?

**Comment by Mandur** [comment]
Thanks for the suggestion @TechPreacher that sound like a lovely idea!

**Comment by Mandur** [comment]
https://dev.azure.com/epicstuff/Azure%20IoT%20Edge%20LoRaWAN%20Starter%20Kit/_build/results?buildId=24621&view=logs&j=7826f347-8899-53d0-eafc-d3fbcaf988d6&t=b3b7d5bb-2197-57f0-8944-1e6878e9d878 All test worked except OTAAJoinTest

**Comment by TechPreacher** [comment]
> Thanks for the suggestion @TechPreacher that sound like a lovely idea!
I'm happy to take this if you are short on time.

**Comment by Mandur** [comment]
That would be great, Do you include it in another PR then?
OTAAJoinTest passed https://dev.azure.com/epicstuff/Azure%20IoT%20Edge%20LoRaWAN%20Starter%20Kit/_build/results?buildId=24622&view=logs&j=7826f347-8899-53d0-eafc-d3fbcaf988d6&t=7826f347-8899-53d0-eafc-d3fbcaf988d6

File: azure-pipeline-integration-test-steps-template.yaml [hunk 1]
```diff
@@ -11,11 +11,6 @@ parameters:
   - ClassCTest
   
 steps:
-- task: UseDotNet@2
```
Comment (fbeltrao) [inline azure-pipeline-integration-test-steps-template.yaml:14]: Should we replace with install .net core 3?

File: LoRaEngine/test/LoRaWanNetworkServer.Test/MessageProcessor_End2End_NoDep_CloudToDeviceMessage_SizeLimit_Should_Accept.cs [hunk 2]
```diff
@@ -97,7 +97,7 @@ public class MessageProcessor_End2End_NoDep_CloudToDeviceMessage_SizeLimit_Shoul
 
             this.LoRaDeviceClient.SetupSequence(x => x.ReceiveAsync(It.IsAny<TimeSpan>()))
                 .ReturnsAsync(cloudToDeviceMessage)
-                .ReturnsAsync(null);
+                .ReturnsAsync<Message>((Message)null);
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWanNetworkServer.Test/MessageProcessor_End2End_NoDep_CloudToDeviceMessage_SizeLimit_Should_Accept.cs:100]: I think you can do either ReturnsAsync<Message>(null) or ReturnsAsync((Message)null)

File: LoRaEngine/test/XunitRetryHelper/XunitRetryHelper.csproj [hunk 3]
```diff
@@ -16,4 +16,10 @@
     <CodeAnalysisRuleSet>../../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/XunitRetryHelper/XunitRetryHelper.csproj:24]: We should have those in the `stylecop.props` file, this way they are shared among all projects.

File: LoRaEngine/test/LoraKeysManagerFacade.Test/LoraKeysManagerFacade.Test.csproj [hunk 4]
```diff
@@ -27,5 +30,11 @@
     <CodeAnalysisRuleSet>../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/LoraKeysManagerFacade.Test/LoraKeysManagerFacade.Test.csproj:38]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/test/LoRaWan.Test.Shared/LoRaWan.Test.Shared.csproj [hunk 5]
```diff
@@ -28,5 +28,11 @@
     <CodeAnalysisRuleSet>../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWan.Test.Shared/LoRaWan.Test.Shared.csproj:37]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/test/LoRaWan.SimulatedTest/LoRaWan.SimulatedTest.csproj [hunk 6]
```diff
@@ -35,5 +38,11 @@
     <CodeAnalysisRuleSet>../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWan.SimulatedTest/LoRaWan.SimulatedTest.csproj:46]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/LoraKeysManagerFacade/LoraKeysManagerFacade.csproj [hunk 7]
```diff
@@ -39,4 +38,10 @@
     <CodeAnalysisRuleSet>../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/LoraKeysManagerFacade/LoraKeysManagerFacade.csproj:46]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/test/LoRaWan.IntegrationTest/LoRaWan.IntegrationTest.csproj [hunk 8]
```diff
@@ -53,5 +52,11 @@
     <CodeAnalysisRuleSet>../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWan.IntegrationTest/LoRaWan.IntegrationTest.csproj:60]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaTools.csproj [hunk 9]
```diff
@@ -24,5 +24,11 @@
     <CodeAnalysisRuleSet>../../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaTools.csproj:32]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/Logger.csproj [hunk 10]
```diff
@@ -16,5 +16,11 @@
     <CodeAnalysisRuleSet>../../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/Logger.csproj:24]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaWan.NetworkServer.csproj [hunk 11]
```diff
@@ -30,5 +24,11 @@
     <CodeAnalysisRuleSet>../../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/LoRaWan.NetworkServer.csproj:32]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaWanTest.csproj [hunk 12]
```diff
@@ -27,5 +30,11 @@
     <CodeAnalysisRuleSet>../../../../stylecop.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
   <Import Project="../../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaToolsTest/LoRaWanTest.csproj:38]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: LoRaEngine/test/LoRaWanNetworkServer.Test/LoRaWanNetworkServer.Test.csproj [hunk 13]
```diff
@@ -30,4 +30,10 @@
     <AdditionalFiles Include="../../../stylecop.json" Link="stylecop.json" />
   </ItemGroup>
   <Import Project="../../../stylecop.props" />
+  <ItemGroup>
+    <PackageReference Update="StyleCop.Analyzers" Version="1.1.118">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
```
Comment (fbeltrao) [inline LoRaEngine/test/LoRaWanNetworkServer.Test/LoRaWanNetworkServer.Test.csproj:38]: We should have those in the stylecop.props file, this way they are shared among all projects.

File: azure-pipeline-integration-test-steps-template.yaml (hunk 1 above)
Comment (Mandur) [reply azure-pipeline-integration-test-steps-template.yaml:14]: I needed to add this steps as .NET 2 is not present on all azure devops machines (same problem that we faced in the current project). .NET 3 is installed by default

File: LoRaEngine/test/LoraKeysManagerFacade.Test/LoraKeysManagerFacade.Test.csproj (hunk 4 above)
Comment (Mandur) [reply LoRaEngine/test/LoraKeysManagerFacade.Test/LoraKeysManagerFacade.Test.csproj:38]: That might explain why Stylecop got so aggressive :p

**Pull request merged**